app/src/main/java/net/gsantner/opoc/util/GsContextUtils.java

```diff
@@ -2911,25 +2911,24 @@ public static void stopBlinking(final View view) {
 
     public static boolean fadeInOut(final View in, final View out, final boolean animate) {
         // Do nothing if we are already in the correct state
-        if (in.getVisibility() == View.VISIBLE && out.getVisibility() == View.GONE) {
+        if (in.getVisibility() == View.VISIBLE && out.getVisibility() == View.INVISIBLE) {
             return false;
         }
 
-        in.setVisibility(View.VISIBLE);
         if (animate) {
-            in.setAlpha(0);
-            in.animate().alpha(1).setDuration(200).setListener(null);
             out.animate()
-                    .alpha(0)
-                    .setDuration(200)
-                    .setListener(new AnimatorListenerAdapter() {
-                        @Override
-                        public void onAnimationEnd(Animator animation) {
-                            out.setVisibility(View.GONE);
-                        }
-                    });
+                    .alpha(0f)
+                    .setDuration(300)
+                    .withEndAction(() -> out.setVisibility(View.INVISIBLE));
+
+            in.setAlpha(0f);
+            in.setVisibility(View.VISIBLE);
+            in.animate()
+                    .alpha(1f)
+                    .setDuration(300);
```

Comment:
Improving the transition and changing from gone to invisible -> dramatically improves text layout perfromance